codegen/service/service_data.go

```diff
@@ -288,6 +364,10 @@ func (d ServicesData) analyze(service *design.ServiceExpr) *Data {
 				ratt = ut.Attribute()
 			}
 			types = append(types, collectTypes(ratt, seen, scope)...)
+			if isProjectable(m.Result.Type) {
+				_, pts := collectProjectedTypes(m.Result, seenProj, scope, viewspkg)
+				projTypes = append(projTypes, pts...)
```

Comment:
Could that produce duplicates in `projTypes`? Does it matter?

Reply:
it won't produce duplicates since `seenProj` will stop it from adding it to the return value.